### PR TITLE
Feat: coinjoin liquidity clue

### PR DESCRIPTION
**Pull request by szymonlesisz**
## Description
Version 3 of coinjoin middleware with added methods:
- /init-liquidity-clue
- /update-liquidity-clue
- /get-liquidity-clue

cc @vdovhanych pls sign updated binaries

`liquidityClue` is used by utxos+round selection algorithm (middleware /select-inputs-for-round) and it's meant to prevent loosing anonymity by giving less priority for utxos with higher anonymity levels.

`liquidityClue` is calculated in two cases:
- on initial account loading. value is calculated from tx history recent coinjoin transaction
> calculation is less precise because of unknown `maxSuggestedAmount` value
> it is stored in CoinjoinAccount is suite, next to `targetAnonymity` as this value not related to session but rather to account
- on each coinjoin tx sigining. value is calculated from currently signed transaction
> value is updated regardless if transaction is broadcasted or not

additional change:
while i was changing `analyzeTransactions` to get initial `liquidityClue` i've also disabled fallback for anonymity levels.
if it fails, then account (or only part of it) should not be available since it's critical to maintain account privacy level (attending coinjoins, send form etc)

### CODE REVIEW

**Comment by szymonlesisz** [comment]
im putting this to draft, beacuse i want to test it before it gets to develop

but code changes are ready to review

File: packages/coinjoin/src/client/analyzeTransactions.ts [hunk 1]
```diff
@@ -58,13 +58,33 @@ export const analyzeTransactions = async (
...
-    return result.reduce((dict, { address, anonymitySet }) => {
+    // find most recent coinjoin transaction in history
+    const cjTx = transactions.find(tx => tx.type === 'joint');
+    let rawLiquidityClue: middleware.RawLiquidityClue = null;
+    if (cjTx) {
+        const [_, externalOutputs] = arrayPartition(
+            cjTx.details.vout.map(vout => transformVinVout(vout, options.network)),
+            vout => 'address' in vout,
+        );
+        const externalAmounts = externalOutputs.map(o => Number(o.value));
```
Comment (marekrjpolak) [inline packages/coinjoin/src/client/analyzeTransactions.ts]: Is this just `filter`? What about https://github.com/trezor/trezor-suite/pull/7231/commits/bfc1ee22fd6e3398f01b99a617d0f29b712d6022?

File: packages/suite/src/reducers/wallet/coinjoinReducer.ts [hunk 2]
```diff
@@ -263,6 +274,9 @@ export const coinjoinReducer = (
             case COINJOIN.ACCOUNT_CREATE:
                 createAccount(draft, action.payload);
                 break;
+            case COINJOIN.ACCOUNT_SET_LIQUIDITY_CLUE:
```
Comment (marekrjpolak) [inline packages/suite/src/reducers/wallet/coinjoinReducer.ts:277]: How often is coinjoin account stored into DB? Shouldn't it be stored after this action as well?

File: packages/suite/src/actions/wallet/coinjoinAccountActions.ts [hunk 3]
```diff
@@ -235,17 +249,27 @@ export const fetchAndUpdateAccount =
                 // TODO accountInfo.utxo don't have proper utxo.confirmations field, only 0/1
 
                 // calculate account anonymity set in CoinjoinClient
-                const accountInfoWithAnonymitySet = await dispatch(
-                    analyzeTransactions(accountInfo, account.symbol),
-                );
+                const result = await dispatch(analyzeTransactions(accountInfo, account.symbol));
+                if (!result) {
```
Comment (marekrjpolak) [inline packages/suite/src/actions/wallet/coinjoinAccountActions.ts]: This error could be handled as the other ones. What about https://github.com/trezor/trezor-suite/pull/7231/commits/e2366afcce61cac928b0e69852ce5159b65d2758?

Comment (szymonlesisz) [reply packages/suite/src/actions/wallet/coinjoinAccountActions.ts]: fair point, i've created followup issue for that
https://github.com/trezor/trezor-suite/issues/7247

File: packages/coinjoin/src/client/analyzeTransactions.ts (hunk 1 above)
Comment (szymonlesisz) [reply packages/coinjoin/src/client/analyzeTransactions.ts]: cool, thx

File: packages/suite/src/reducers/wallet/coinjoinReducer.ts (hunk 2 above)
Comment (szymonlesisz) [reply packages/suite/src/reducers/wallet/coinjoinReducer.ts:277]: good catch, it should

Comment (szymonlesisz) [reply packages/suite/src/reducers/wallet/coinjoinReducer.ts:277]: https://github.com/trezor/trezor-suite/pull/7231/commits/ef5072db0064f8adfef030c46f53fb9992efc5fb

**Comment by szymonlesisz** [comment]
virtually approving the rest :) i cannot approve my own PR

it was tested today and looks like its working, so we can merge it

**Review by marekrjpolak — APPROVED** [review]

**Pull request merged**